src/Integrations/IntegrationRegistry.php

```diff
@@ -0,0 +1,179 @@
+<?php
+namespace Automattic\WooCommerce\Blocks\Integrations;
+
+/**
+ * Class used for tracking registered integrations with various Block types.
+ */
+class IntegrationRegistry {
+	/**
+	 * Integration identifier is used to construct hook names and is given when the integration registry is initialized.
+	 *
+	 * @var string
+	 */
+	protected $registry_identifier = '';
+
+	/**
+	 * Registered integrations, as `$name => $instance` pairs.
+	 *
+	 * @var IntegrationInterface[]
+	 */
+	protected $registered_integrations = [];
+
+	/**
+	 * Initializes all registered integrations.
+	 *
+	 * Integration identifier is used to construct hook names and is given when the integration registry is initialized.
+	 *
+	 * @param string $registry_identifier Identifier for this registry.
+	 */
+	public function initialize( $registry_identifier = '' ) {
+		if ( $registry_identifier ) {
+			$this->registry_identifier = $registry_identifier;
+		}
+
+		if ( empty( $this->registry_identifier ) ) {
+			_doing_it_wrong( __METHOD__, esc_html( __( 'Integration registry requires an identifier.', 'woo-gutenberg-products-block' ) ) );
+			return false;
+		}
+
+		/**
+		 * Hook: integration_registration.
+		 *
+		 * Runs before integrations are initialized allowing new integration to be registered for use. This should be
+		 * used as the primary hook for integrations to include their scripts, styles, and other code extending the
+		 * blocks.
+		 *
+		 * @param IntegrationRegistry $this Instance of the IntegrationRegistry class which exposes the IntegrationRegistry::register() method.
+		 */
+		do_action( 'woocommerce_blocks_' . $this->registry_identifier . '_registration', $this );
```

Comment:
Could we only send `::register()` and not the entire class instance?

Reply:
This is matching the pattern in the Payments Registry. There are some other useful methods like unregistration that may be needed.